### DISCUSSION TITLE
Type drop down -  does it work?

### DISCUSSION THREAD

**Discussion by step21** [louisgv/local.ai#68] [Q&A] [answered]
I noticed that the models show the type, but that this actually is a dropdown. I was expecting this to be a field, like `tokenizer`. Are there actually instances where it is possible to change the type there?

**Comment by louisgv** [comment, marked as answer]
@step21 the types are deterministic atm since the backend only support those model type. If you would like to add additional type, the steps are:

1. Add that model into the backend (either as a top-up implementation/trait of local.ai backend OR as a PR to the [llm rust crate](https://github.com/rustformers/llm/) 
2. Add that model as an enum: https://github.com/louisgv/local.ai/blob/main/models/_shared.ts#L1-L8
3. Add a known model with that type
